module/state_synchronization/requester/jobs/execution_data_reader.go

```diff
@@ -94,7 +97,13 @@ func (r *ExecutionDataReader) getExecutionData(signalCtx irrecoverable.SignalerC
 		return nil, fmt.Errorf("failed to lookup header for height %d: %w", height, err)
 	}
 
-	result, err := r.results.ByBlockID(header.ID())
+	// get the ExecutionResultID for the block from the block's seal
+	seal, err := r.seals.BySealedBlockID(header.ID())
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup seal for block %s: %w", header.ID(), err)
+	}
+
+	result, err := r.results.ByID(seal.ResultID)
```

Comment:
nit: we could extract this into a function, since we don't care about the seal but only the result